ironfish/src/storage/levelup/store.ts

```diff
@@ -75,19 +75,32 @@ export class LevelupStore<Schema extends DatabaseSchema> extends DatabaseStore<S
     transaction?: IDatabaseTransaction,
     keyRange?: DatabaseKeyRange,
   ): AsyncGenerator<[SchemaKey<Schema>, SchemaValue<Schema>]> {
-    if (keyRange) {
-      keyRange = StorageUtils.addPrefixToRange(keyRange, this.prefixBuffer)
+    if (keyRange?.gt || keyRange?.gte || keyRange?.lt || keyRange?.lte) {
+      keyRange = {
+        ...keyRange,
+        ...StorageUtils.addPrefixToRange(keyRange, this.prefixBuffer),
+      }
```

Comment:
I think `StorageUtils.addPrefixToRange` should change so that it propagates `ordered` and `reverse` if they're set.

Reply:
Made a change in another PR that removes the need for this change - https://github.com/iron-fish/ironfish/pull/2798